server/lib/person.js

```diff
@@ -1,39 +1,33 @@
 const databasePerson = require('../services/person')
 
 class Person {
-  static object (data) {
-    return {
-      personId: data.personid,
-      staffId: data.staffid,
-      firstName: data.firstname,
-      surname: data.surname,
-      email: data.email,
-      gradeId: data.gradeid,
-      locationId: data.locationid,
-      managerId: data.managerid,
-      role: data.role,
-      statusId: data.statusid,
-      skills: [],
-      archived: data.archived
-    }
+  constructor(data) {
+    this.personId = data.personid
+    this.staffId = data.staffid
+    this.firstName = data.firstname
+    this.surname = data.surname
+    this.email = data.email
+    this.gradeId = data.gradeid
+    this.locationId = data.locationid
+    this.managerId = data.managerid
+    this.role = data.role
+    this.statusId = data.statusid
+    this.skills = []
+    this.archived = data.archived
   }
 
-  static buildObject (data) {
-    let person
-    data.forEach(row => {
-      if (row.kind === 'details') {
-        person = this.object(row)
-      } else {
-        person.skills.push(row.skillid)
-      }
-    })
+  static fromRows(data) {
+    const personRow = data.find(r => r.kind === 'details')
+    const person = personRow ? new Person(personRow) : {}
+    person.skills = data.filter(r => r.kind === 'skill').map(s => s.skillid)
+
     return person
   }
 
-  static async getPerson (personId) {
+  static async getPerson(personId) {
     const query = `WHERE personid = ${personId}`
     const result = await databasePerson.getPerson(query)
-    return this.buildObject(result.rows)
+    return Person.fromRows(result.rows)
```

Comment:
renamed the function to reveal the source data
It's easer to tell it's a static method if you use the class name rather than 'this'